src/server/system_services/pool_server.js

```diff
@@ -767,37 +554,31 @@ async function delete_hosts_pool(req, pool) {
         });
     }
 
-    pool_scaling_sem.surround_key(pool.name, async () => {
-        if (host_count > 0) {
-            await nodes_client.instance()
-                .delete_hosts_by_pool(pool.name, req.system._id);
+    if (host_count > 0) {
+        await nodes_client.instance()
+            .delete_hosts_by_pool(pool.name, req.system._id);
+    }
+    dbg.log0(`delete_hosts_pool: removing pool ${pool.name} from the database`);
+    await system_store.make_changes({
```

Comment:
You cannot remove from the DB before all the agents are removed. If you do so and the operator is failing for some reason the core will have agents connected that are not part of any pool. 
Removing the pool from the database be done as a result of an operation the operator should do when it verified that all pods are down

Reply:
if it will fail it will throw an exception - we won't update the db